lib/actions/user.js

```diff
@@ -49,44 +49,56 @@ export function fetchUserMonitoredTrips (accessToken) {
     const { otp } = getState()
     const { otp_middleware: otpMiddleware = null } = otp.config.persistence
 
     if (otpMiddleware) {
-      const { user: authUser } = auth0
-      const { data: user, status: fetchUserStatus } = await fetchUser(otpMiddleware, accessToken)
-
-      // Beware! On AWS API gateway, if a user is not found in the middleware
-      // (e.g. they just created their Auth0 password but have not completed the account setup form yet),
-      // the call above will return, for example:
-      // {
-      //    status: 'success',
-      //    data: {
-      //      "result": "ERR",
-      //      "message": "No user with id=000000 found.",
-      //      "code": 404,
-      //      "detail": null
-      //    }
-      // }
-      //
-      // The same call to a middleware instance that is not behind an API gateway
-      // will return:
-      // {
-      //    status: 'error',
-      //    message: 'Error get-ing user...'
-      // }
-      // TODO: Improve AWS response.
-
-      const isNewAccount = fetchUserStatus === 'error' || (user && user.result === 'ERR')
-      if (!isNewAccount) {
-        // Load user's monitored trips before setting the user state.
-        await dispatch(fetchUserMonitoredTrips(accessToken))
+      // First, get the Auth0 access token.
+      let accessToken
+      try {
+        accessToken = await auth0.getAccessTokenSilently()
+      } catch (error) {
+        // TODO: improve UI if there is an errror.
+        alert('Error obtaining an authorization token.')
+      }
 
-        dispatch(setCurrentUser({ accessToken, user }))
-      } else {
-        dispatch(setCurrentUser({ accessToken, user: createNewUser(authUser) }))
+      // Once accessToken is available, proceed to fetch or initialize loggedInUser.
+      if (accessToken) {
+        const { user: authUser } = auth0
+        const { data: user, status: fetchStatus } = await fetchUser(otpMiddleware, accessToken)
+
+        // Beware! On AWS API gateway, if a user is not found in the middleware
+        // (e.g. they just created their Auth0 password but have not completed the account setup form yet),
+        // the call above will return, for example:
+        // {
+        //    status: 'success',
+        //    data: {
+        //      "result": "ERR",
+        //      "message": "No user with id=000000 found.",
+        //      "code": 404,
+        //      "detail": null
+        //    }
+        // }
+        //
+        // The same call to a middleware instance that is not behind an API gateway
+        // will return:
+        // {
+        //    status: 'error',
+        //    message: 'Error get-ing user...'
+        // }
+        // TODO: Improve AWS response.
+
+        const isNewAccount = fetchStatus === 'error' || (user && user.result === 'ERR')
+        if (!isNewAccount) {
+          // Load user's monitored trips before setting the user state.
+          await dispatch(fetchUserMonitoredTrips(accessToken))
```

Comment:
Sorry I didn't flag this before, but why does the order matter here?

Reply:
We want the user's trips to be available when `state.user.loggedInUser` is set, so other parts of the UI don't have to check whether the monitored trips have been loaded or not. This would be most noticeable when you log in directly from the `/#/savedtrips` route.

Reply:
I added a comment regarding that in https://github.com/opentripplanner/otp-react-redux/pull/233/commits/c9bb706107fffe6b477e73123313da05a5031f74.

Reply:
What parts of the UI are fetching the monitored trips? I couldn't really find where this would happen. In any case, this doesn't seem like a pattern we want to embed here. In my view we want to update the store with the user as soon as possible so that we're not waiting for an extra request to the backend to be fulfilled. Also, it avoids calling `serCurrentUser` multiple times.

Reply:
See my comment about setting a default state in the user reducer. I don't think we should be relying on this order of operations for determining whether a user is allowed to create a new monitored trip.

Reply:
If we want to remove fetching trips when fetching the user data, then `state.user.loggedInUserMonitoredTrips` needs to be fetched on first use from `SavedTripList` and `SavedTripScreen`. Each of these components will need to invoke the fetch and display a hint while fetching is taking place. Is that aligned with your thinking @landonreed ?

Reply:
I changed the loading of user monitored trips in https://github.com/opentripplanner/otp-react-redux/pull/233/commits/99755e59acd2b46332998b7c656436acff555919. It wasn't too big of a change this time, otherwise it would go in a separate PR.